app/src/main/java/space/taran/arkmemo/data/repositories/VersionStorage.kt

```diff
@@ -0,0 +1,139 @@
+                    Version(Version.Content(listOf()), meta)
+                } else {
+                    val oldVerContent =
+                        JsonParser.parseVersionContentFromJson(jsonVersion.toString())
+                    Version(oldVerContent, meta)
+                }
+            val idListMutable = verToWrite.content.idList.toMutableList()
+            idListMutable.add(id)
+            writeToVersionFile(versionFile, Version.Content(idListMutable.toList()))
+        }
+    }
+
+    fun removeVersion(id: ResourceId, removedVersion: ResourceId): Int {
+        val versionContentNow = getVersionContent(removedVersion)
+        val notePath = root.resolve("$id.${TextNotesRepository.NOTE_EXT}")
+        if (versionContentNow.isEmpty()) {//means that root TextNote its being deleted, and no more content
+            //just delete note and version.
+            removeFileFromMemory(notePath)
+            removeFileFromMemory(versionPath(id))
+            return CODES_DELETING_NOTE.SUCCESS_NOTE_AND_VERSION_DELETED.code
+        } else if (removedVersion == id) {//means that root TextNote its being deleted,
+            val secondNoteFromVId = versionContentNow[0]
+            val versionFile = versionPath(removedVersion).toFile()
+            val newVersionFile = versionPath(secondNoteFromVId).toFile()
+            //rename version file:
+            versionFile.renameTo(newVersionFile)
+            val newVersionContent = Version.Content(versionContentNow.drop(1))
+            writeToVersionFile(newVersionFile, newVersionContent)
+            //delete note:
+            removeFileFromMemory(notePath)
+            return CODES_DELETING_NOTE.SUCCESS_NOTE_DELETED_VERSION_CHANGED.code
+        } else {
+            val noteIndex = versionContentNow.indexOf(id)
+            if (noteIndex == -1) {
+                return CODES_DELETING_NOTE.NOTE_NOT_DELETED.code
+            }
+            val idListMutable = versionContentNow.toMutableList()
+            idListMutable.removeAt(noteIndex)
+            //overwrite version file:
+            writeToVersionFile(
+                versionPath(removedVersion).toFile(),
+                Version.Content(idListMutable.toList())
+            )
+            //delete note:
+            removeFileFromMemory(notePath)
+        }
+        return CODES_DELETING_NOTE.SUCCESS.code
+    }
+
+    private fun removeFileFromMemory(path: Path) {
+        Files.deleteIfExists(path)
+    }
+
+    private fun getVersionContent(rootResourceId: ResourceId): List<ResourceId> {
+        val versionPath = versionPath(rootResourceId)
+        var verContent: Version.Content? = null
+        try {
+            val fileReader = FileReader(versionPath.toFile())
+            val bufferedReader = BufferedReader(fileReader)
+            val jsonVersion = StringBuilder()
+            with(bufferedReader) {
+                forEachLine {
```

Comment:
Seems that we have 4 instances of this code reading a file, we should have a method for this.